addon/addon/models/orchestration-matrix.ts

```diff
@@ -0,0 +1,207 @@
+import { Frame } from 'animations-experiment/behaviors/base';
+import Sprite, { MotionProperty } from './sprite';
+import {
+  MotionDefinition,
+  ParallelAnimationTimeline,
+  SequentialAnimationTimeline,
+} from './transition-runner';
+
+interface RowFragment {
+  startColumn: number;
+  frames: Frame[];
+}
+
+export class OrchestrationMatrix {
+  constructor(
+    // This isn't ordered yet, but can be ordered if necessary
+    public rows = new Map<Sprite, RowFragment[]>(),
+    public totalColumns = 0
+  ) {}
+
+  // TODO: if we want to handle cascading effects, we'll need a way of iterating over this column-by-column,
+  // in order of sprites from higher level in DOM to lower level, modifying scoped variables per-sprite
+  // TODO: this can probably become a generator if we want?
+  getKeyframes(
+    constructKeyframe: (
+      previousKeyframe: Partial<Keyframe>,
+      frames: Frame[] // frame order may be relevant
+    ) => Keyframe
+  ) {
+    let result: Map<Sprite, Keyframe[]> = new Map();
+    for (let [sprite, rowFragments] of this.rows) {
+      // convenience so we do less operations to determine active fragments
+      let fragmentsByColumn: Record<number, RowFragment[]> = {};
+      // examine assumptions - what exactly is the frame?
+      let baseKeyframe = {};
```

Comment:
Backfilling of keyframe state is currently done by creating `baseKeyframe`, here I'm making the assumption that the properties in the first frame are all the properties that will be manipulated by that set of frames. So a set of keyframes that is manipulating `transform` and `opacity` will have `transform` and `opacity` both set in its first frame, it won't have only `opacity` while `transform` comes in later.